shared/src/business/useCases/getPractitionerCasesInteractor.ts

```diff
@@ -30,12 +32,34 @@ export const getPractitionerCasesInteractor = async (
     .getPersistenceGateway()
     .getCasesByDocketNumbers({ applicationContext, docketNumbers });
 
-  cases.forEach(
-    aCase => (aCase.caseTitle = Case.getCaseTitle(aCase.caseCaption)),
-  );
+  const caseDetails: PractitionerCaseDetail[] = cases.map(c => {
+    const formattedCase = formatCase(applicationContext, c, authorizedUser);
+
+    const {
+      caseTitle,
+      consolidatedIconTooltipText,
+      docketNumber,
+      docketNumberWithSuffix,
+      inConsolidatedGroup,
+      isLeadCase,
+      isSealed,
+      status,
+    } = formattedCase;
+
+    return {
+      caseTitle,
+      consolidatedIconTooltipText,
+      docketNumber,
+      docketNumberWithSuffix,
+      inConsolidatedGroup,
+      isLeadCase,
+      isSealed,
+      status,
+    };
+  });
```

Comment:
Before, we were returning a whole "case" object. (I put "case" in quotes because the available properties on a case differ depending on where it is at.) I have restricted this for two reasons:

1. The restriction (hopefully) prevents a 502 error when there are a lot of cases. See [troubleshooting doc](https://github.com/ustaxcourt/ef-cms/blob/staging/docs/additional-resources/troubleshooting.md#502-errors).
2. The restriction makes exactly what we have/need for a practitioner case explicit.

An alternative approach would be to paginate here on the backend, but that would introduce further complexities, especially if we wanted to avoid re-sorting to get the right records on every page change.

Reply:
Do you happen to have an estimate of how much more "considerably" more would be for this? And how performant is this now for the 4000-case petitioner? 

I like the limit on what's returned regardless of choice to paginate on the backend, but I'm curious how long we will have before pagination becomes a thing we have to worry about again, either due to 502 errors or loading times -- 1 month? 12 months? Several years? This will probably be a tough question to get a precise answer on but I am curious if we can come up with a super rough estimate.

Reply:
Good questions!

The current 502-causing practitioner has ~4300 cases. Based on some local experiments (jsonifying the response and getting blob size), without the filtering we get a ~10.85 MB response size, well over the 6 MB lambda limit. With filtering, we get a ~.8 MB response size, well under. Even with these rough calculations, this seems to indicate enough wiggle room to handle any practical case list size.

Performance-wise, things seem ok. For one, most users have far, far fewer cases, so this is not a common occurrence. For two, other users with 1000+ cases were snappy enough. (That said, I still need to test in the test environment to see the impact of a non-linear slowdown--we do sort, for instance--for this particular user.)

Reply:
Thanks for the good questions!